client/add.go

```diff
@@ -22,16 +31,33 @@ For more about knox, see https://github.com/pinterest/knox.
 See also: knox create, knox promote
 	`,
 }
+var addTinkKeyset = cmdAdd.Flag.String("key-template", "", "")
 
 func runAdd(cmd *Command, args []string) {
 	if len(args) != 1 {
 		fatalf("add takes only one argument. See 'knox help add'")
 	}
-	fmt.Println("Reading from stdin...")
 	keyID := args[0]
-	data, err := ioutil.ReadAll(os.Stdin)
+	var data []byte
+	var err error
+	if *addTinkKeyset != "" {
+		templateName := *addTinkKeyset
+		err = checkTemplateNameAndKnoxIDForTinkKeyset(templateName, keyID)
```

Comment:
nit: these function names are really long. if you can think of a more concise way of saying the same thing it might be better.

Reply:
Thank you so much for your comment! I have revised the code.